.github/workflows/vercel.yml

```diff
@@ -0,0 +1,51 @@
+name: Vercel
+
+on:
+  workflow_call:
+    inputs:
+      env_name:
+        description: 'Environment to deploy to. Options are: dev, staging, barn and prod'
+        required: true
+        type: string
+
+env:
+  VERCEL_ORG_ID: ${{ secrets.VERCEL_ORG_ID }}
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.env_name }} # Environment rules defined on GH UI
+    concurrency: ${{ inputs.env_name }} # Only one run per env at a time
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set VERCEL_PROJECT_ID env var
+        # It's set in each env's config on https://github.com/cowprotocol/cowswap/settings/environments
+        run: echo "VERCEL_PROJECT_ID=${{ secrets.VERCEL_PROJECT_ID }}" >> $GITHUB_ENV
+
+      - name: Install Vercel CLI
+        run: npm install --global vercel@latest
+
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=production --token=${{ secrets.VERCEL_TOKEN }}
+
+      - name: Build Project Artifacts
+        run: >
+          REACT_APP_SENTRY_DSN=${{ secrets.SENTRY_DSN }}
+          REACT_APP_SENTRY_AUTH_TOKEN=${{ secrets.SENTRY_AUTH_TOKEN }}
+          GOOGLE_ANALYTICS_ID=${{ secrets.GOOGLE_ANALYTICS_ID }}
+          APP_ID=1
```

Comment:
this is how we share envs from GH to vercel?

Reply:
Yep!

Actually, the env vars don't go to Vercel.
They are used to create the build locally and only the generated files are updated there